Flaky test in `ElasticJobExecutorTest` due to HashMap ordering

## Bug Report

### Which version of ElasticJob did you use?
Latest master branch (commit: 4acbf77d3fc05f3c4f484c38b06d6aed786e5974)
### Expected behavior
The test `ElasticJobExecutorTest.assertExecuteFailureWhenThrowExceptionForMultipleShardingItems` should pass consistently regardless of the internal ordering of HashMap entries.
### Actual behavior

The test fails intermittently with the following error:

```[ERROR] ElasticJobExecutorTest.assertExecuteFailureWhenThrowExceptionForMultipleShardingItems:127->assertExecuteFailureWhenThrowException:138 
Argument(s) are different! Wanted:
jobFacade.postJobStatusTraceEvent(
    "fake_task_id",
    TASK_ERROR,
    "{0=java.lang.RuntimeException
, 1=java.lang.RuntimeException
}"
);

Actual invocations have different arguments:
jobFacade.postJobStatusTraceEvent(
    "fake_task_id",
    TASK_ERROR,
    "{1=java.lang.RuntimeException
, 0=java.lang.RuntimeException
}"
);
```
### Reason analyze (If you can)
In `ElasticJobExecutor.java`, the `itemErrorMessages` is a `ConcurrentHashMap<Integer, String>` (initialized in constructor):

When this map is converted to a string via `toString()`, the iteration order is non-deterministic. This causes the error message format to vary between executions (e.g., {0=..., 1=...} or {1=..., 0=...}).

The test method `getErrorMessage` only generates the expected string with a fixed order {0=..., 1=...}, which fails when the HashMap's internal ordering produces {1=..., 0=...}.

### Steps to reproduce the behavior.

Run the test multiple by using [NonDex](https://github.com/TestingResearchIllinois/NonDex):

`mvn -pl kernel edu.illinois:nondex-maven-plugin:2.2.1:nondex  -DnondexRuns=10`
The test will fail intermittently depending on the HashMap's internal ordering

--- a/kernel/src/test/java/org/apache/shardingsphere/elasticjob/kernel/executor/ElasticJobExecutorTest.java
+++ b/kernel/src/test/java/org/apache/shardingsphere/elasticjob/kernel/executor/ElasticJobExecutorTest.java
@@ -42,6 +42,7 @@ import java.util.Map;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.argThat;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.lenient;
@@ -135,17 +136,21 @@ class ElasticJobExecutorTest {
         } finally {
             verify(jobFacade).postJobStatusTraceEvent(shardingContexts.getTaskId(), State.TASK_STAGING, "Job 'test_job' execute begin.");
             verify(jobFacade).postJobStatusTraceEvent(shardingContexts.getTaskId(), State.TASK_RUNNING, "");
-            verify(jobFacade).postJobStatusTraceEvent(shardingContexts.getTaskId(), State.TASK_ERROR, getErrorMessage(shardingContexts));
+            verify(jobFacade).postJobStatusTraceEvent(eq(shardingContexts.getTaskId()), eq(State.TASK_ERROR), argThat(msg -> isValidErrorMessage(msg, shardingContexts)));
             verify(jobFacade).registerJobBegin(shardingContexts);
             verify(jobItemExecutor, times(shardingContexts.getShardingTotalCount())).process(eq(fooJob), eq(jobConfig), eq(jobRuntimeService), any());
             verify(jobFacade).registerJobCompleted(shardingContexts);
         }
     }
     
-    private String getErrorMessage(final ShardingContexts shardingContexts) {
-        return 1 == shardingContexts.getShardingItemParameters().size()
-                ? "{0=java.lang.RuntimeException" + System.lineSeparator() + "}"
-                : "{0=java.lang.RuntimeException" + System.lineSeparator() + ", 1=java.lang.RuntimeException" + System.lineSeparator() + "}";
+    private boolean isValidErrorMessage(final String errorMessage, final ShardingContexts shardingContexts) {
+        if (1 == shardingContexts.getShardingItemParameters().size()) {
+            return errorMessage.equals("{0=java.lang.RuntimeException" + System.lineSeparator() + "}");
+        } else {
+            String pattern1 = "{0=java.lang.RuntimeException" + System.lineSeparator() + ", 1=java.lang.RuntimeException" + System.lineSeparator() + "}";
+            String pattern2 = "{1=java.lang.RuntimeException" + System.lineSeparator() + ", 0=java.lang.RuntimeException" + System.lineSeparator() + "}";
+            return errorMessage.equals(pattern1) || errorMessage.equals(pattern2);
+        }
     }
     
     @Test

--- a/kernel/src/test/java/org/apache/shardingsphere/elasticjob/kernel/executor/ElasticJobExecutorTest.java
+++ b/kernel/src/test/java/org/apache/shardingsphere/elasticjob/kernel/executor/ElasticJobExecutorTest.java
@@ -145,7 +145,7 @@ class ElasticJobExecutorTest {
     
     private boolean isValidErrorMessage(final String errorMessage, final ShardingContexts shardingContexts) {
         if (1 == shardingContexts.getShardingItemParameters().size()) {
-            return errorMessage.equals("{0=java.lang.RuntimeException" + System.lineSeparator() + "}");
+            return ("{0=java.lang.RuntimeException" + System.lineSeparator() + "}").equals(errorMessage);
         } else {
             String pattern1 = "{0=java.lang.RuntimeException" + System.lineSeparator() + ", 1=java.lang.RuntimeException" + System.lineSeparator() + "}";
             String pattern2 = "{1=java.lang.RuntimeException" + System.lineSeparator() + ", 0=java.lang.RuntimeException" + System.lineSeparator() + "}";

